src/pages/Authentication/Signup.jsx

```diff
@@ -1,81 +1,150 @@
-import React from 'react'
+import React from "react";
 import "./auth.css";
-import {Link} from "react-router-dom";
-import { useDocumentTitle } from '../../hooks/useDocumentTitle';
-import FormTextInput from '../../components/FormInputText/FormInputText';
-import { useAuthForm } from '../../hooks/useAuthForm/useAuthForm';
-import axios from 'axios';
-import { useAuth } from '../../contexts/Providers/AuthProvider/AuthProvider';
-import { useEffect } from 'react';
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useDocumentTitle } from "../../hooks/useDocumentTitle";
+import FormTextInput from "../../components/FormInputText/FormInputText";
+import { useAuthForm } from "../../hooks/useAuthForm/useAuthForm";
+import axios from "axios";
+import { useAuth } from "../../contexts/Providers/AuthProvider/AuthProvider";
+import { useEffect } from "react";
+import { toast } from "react-toastify";
 
-const EMAIL_REGEX=/^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$/;
+const EMAIL_REGEX = /^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$/;
 
 function Signup() {
   useDocumentTitle("Signup | CarSmart");
 
-  const {authFormState, authFormDispatch} =useAuthForm();
-  const {authDispatch} = useAuth();
-  const {fullName, email, password, loading, confirmPassword,agreement} = authFormState;
+  const { authFormState, authFormDispatch } = useAuthForm();
+  const { authDispatch } = useAuth();
+  const { fullName, email, password, loading, confirmPassword, agreement } =
+    authFormState;
 
-  useEffect(()=>{
-    return ()=>authFormDispatch({type:"LOADING_OFF"});
-  },[])
+  const location = useLocation();
+  const navigate = useNavigate();
+  useEffect(() => {
+    return () => authFormDispatch({ type: "LOADING_OFF" });
+  }, []);
 
   const signupHandler = async (e) => {
```

Comment:
suggestion : you can move this in separate function so your code looks more cleaner